Avoid save prompt when closing the editor for preview content

DAXaholic opened the pull request:
Currently 'untitled' files are opened when viewing a run, action, trigger or version of a Logic App which causes a save prompt when closing the corresponding editor.
This PR introduces an own content provider for showing read-only files to solve the issue above.

The PR's functionality is more or less 'cherry-picked' from [microsoft/vscode-azuretools](https://github.com/microsoft/vscode-azuretools) as we currently depend on an old version of the package which does not contain the required functionality and updating to a more recent version of it would cause a lot of refactoring. Once we update the dependency to a version which includes the read-only content provider, we should remove this PR's read-only content provider and rewire its callers to use the 'genuine' implementation.

See also #107

joechung-msft commented on src/utils/readOnlyUtils.ts:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    public async provideTextDocumentContent(uri: Uri, token: CancellationToken) {
+        [uri,] = [uri, token]; // avoid warning about ignored token parameter
```
You should be able to prefix `token` with an underscore to avoid the warning about unused parameters: `_token`

joechung-msft commented on src/utils/readOnlyUtils.ts:
```diff
@@ -0,0 +1,59 @@
+*/
+
+import { CancellationToken, Event, EventEmitter, TextDocumentContentProvider, Uri, window, workspace } from "vscode";
+import { createHash} from "crypto";
```
(nit) Insert space between `createHash` and `}`

joechung-msft commented on src/utils/readOnlyUtils.ts:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    public async openReadOnlyContent(name: string, content: string, fileExt: string) {
+        const nameHash  = createHash('sha256').update(name).digest('hex');
```
(nit) Remove extra space between `nameHash` and `=`

joechung-msft commented on src/utils/readOnlyUtils.ts:
```diff
@@ -0,0 +1,59 @@
+
+    public async openReadOnlyContent(name: string, content: string, fileExt: string) {
+        const nameHash  = createHash('sha256').update(name).digest('hex');
+        const uri = Uri.parse(`${providerScheme}:///${nameHash}/${name}${fileExt}`);
```
Is it intentional to have three slashes here instead of two?

DAXaholic replied:
This was brought in from the original [implementation](https://github.com/microsoft/vscode-azuretools/blob/e261027bb70893a08210b81585419a84a5c0d7d4/ui/src/openReadOnlyContent.ts#L51)
So I don't know whether they did it intentionally but I think so

joechung-msft commented on src/utils/readOnlyUtils.ts:
```diff
@@ -0,0 +1,59 @@
+    call sites to the "genuine" implementation.
+*/
+
+import { CancellationToken, Event, EventEmitter, TextDocumentContentProvider, Uri, window, workspace } from "vscode";
```
(nit) Use VSCode's "Organize imports" command to sort lines by import source and imports in each line.

joechung-msft commented on src/utils/readOnlyUtils.ts:
```diff
@@ -0,0 +1,59 @@
+        this._onDidChangeEmitter.fire(uri);
+    }
+
+    public async provideTextDocumentContent(uri: Uri, token: CancellationToken) {
```
Declare return type: `Promise<string>`

joechung-msft commented on src/utils/readOnlyUtils.ts:
```diff
@@ -0,0 +1,59 @@
+        return this._onDidChangeEmitter.event;
+    }
+
+    public async openReadOnlyContent(name: string, content: string, fileExt: string) {
```
Declare return type: `Promise<void>`

DAXaholic commented:
Thanks for review - addressed the feedback.  
Some points came just by copying the stuff from the base repo, at others I had obviously some 'blind moment' yesterday :man_facepalming:
